assets/src/editor/events/hocs/if-valid-date-entity.js

```diff
@@ -0,0 +1,12 @@
+import { createHigherOrderComponent, ifCondition } from '@wordpress/compose';
+import { isModelEntityOfModel } from '@eventespresso/validators';
+
+export default createHigherOrderComponent(
+	ifCondition(
+		( { dateEntity } ) => isModelEntityOfModel(
+			dateEntity,
+			'datetime'
+		)
+	),
+	'ifValidDateEntity'
+);
```

Comment:
This hoc can go inside the `editor/hocs` folder and thus be exported on `@eventespresso/editor-hocs` aliased package.

Also, there's good opportunity to add unit tests for this (which should be straightforward enough).